README.md

```diff
@@ -1,52 +1,167 @@
+
+GitHub exposes a way for us to do this using HTML parameters.  By changing the URL slightly to include the `author` parameter, we can ask the GitHub API to return only the commits made by DHH (bonus points if you know who this is!).
+
+Let's add ```?author=DHH``` to the end of the url and see how the results change.
+
+Go back to Postman and perform the following:
 
-## Intro
+* Enter **https://api.github.com/repos/rails/rails/commits?author=DHH**
+* Click the **Send** button.
 
-So far, when you've been using an API, you've been using a gem to access the API endpoints. Basically, you're relying on work someone else did to access clean data in the desired format.
 
-Instead of relying on a gem or library to do the heavy lifting for us, we can use Ajax to do it ourselves, without having a page refresh. 
+** Maybe link this back to rails or a gem example **
 
+As you can see, getting data from an API is pretty easy but we haven't really said what an API is.
 
-## API Basics
+```ruby
+github_commits = Github::Client::Repos::Commits.new
+github_commits.all(user: 'rails', repo: 'rails')
+github_commits.all(user: 'rails', repo: 'rails', author: 'DHH')
+```
 
-Underneath the hood of all those gems, a request as been made to the API (to a specific url) and then the app responds to that request with the data.
+## Defining API (done)
 
-use postman chrome extension or app?? to make API request to demonstrate getting back JSON and manipulating it (screenshots) from the Github API
+For the purpose of this lesson, we are mostly concerned with web APIs.  But the term API actually has a more broad meaning.
 
-theoretical definition of API
+> In computer programming, an application programming interface (API) is a set of routines, protocols, and tools for building software and applications. - Wikipedia
+
+In its simplest form, an API in relation to Object Oriented programming is a class and the list of methods we define.  When creating a class, we are defining a guidebook on how to interact with other parts of the code.  We get to decide which methods and variables are public or private, essentially controlling how to interact with the class.  When we apply this concept to the web, we get web APIs like the GitHub API and Twitter API. From our Postman experiment, we saw how GitHub provides a way for us interact with the data on their system. Just like how a class provides a set of public methods to interact with, web APIs provide us with urls. The list of urls that GitHub provides on https://developer.github.com/v3 act as the public methods into their system.  The developers that created the API control which resources they want to share and who has access to them. 
 
 ## Ajax
 
-describe what ajax is and what it's used for
+*** SOMETHING HERE to make it flow better ***
```

Comment:
Maybe note here that these APIs are effectively all of the same data that's on github.com or twitter.com or whatever. The key is that it's *only* the data. As a developer of an application we just need the data, we don't need any of the formatting that comes with the HTML/CSS.

As far as flow I wouldn't jump right into AJAX. Present the problem first, then say "wouldn't it be nice if we didn't have the user have to refresh? Wouldn't it be nice if we could do two things at once? While the user is entering data into a form, we could be searching for them in the background. That's where a term called AJAX comes in. 

As far as **blank application** they've made a simple blog before so. Remember in your blog application from previous labs when you had to create a list of posts?